Source/Core/VideoCommon/GraphicsModSystem/Config/GraphicsTarget.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <optional>
+#include <string>
+#include <variant>
+
+#include <picojson.h>
+
+#include "Common/CommonTypes.h"
+#include "VideoCommon/TextureDecoder.h"
+#include "VideoCommon/XFMemory.h"
+
+struct TextureTarget
+{
+  TextureTarget() = default;
+  virtual ~TextureTarget() = default;
+  TextureTarget(const TextureTarget&) = default;
+  TextureTarget(TextureTarget&&) = default;
+  TextureTarget& operator=(const TextureTarget&) = default;
+  TextureTarget& operator=(TextureTarget&&) = default;
```

Comment:
[This is rule of 5](https://en.cppreference.com/w/cpp/language/rule_of_three).  Since the destructor has to be specified (because this is a class with a virtual destructor), it is good practice to specify the others because otherwise implicit move constructor / move assignment operators are not defined.

Reply:
Does it need to be specified, though?  Wouldn't the default one work (since you're using `= default`)?  (Looking at [cppreference](https://en.cppreference.com/w/cpp/language/destructor) it seems like if you use `= default` or leave it out entirely it'll be not user provided, but I'm not sure whether `virtual` matters here.

If `TextureTarget` needs one, then why doesn't `ProjectionTarget` need one?

Reply:
So basically, I consider all this "good practice" but could be argued either way.  You probably know those rules but maybe make different judgements on when to partake in them.

A virtual destructor is only needed if a class is polymorphic and is going to be deleted to clear up memory.

As per cppreference, if a destructor is added (or a copy constructor / assignment operator), then move operations will be implicitly not specified.

Both of these can lead to unexpected bugs in specific situations.  So rather than think about when to add this or to react when issues arise from not doing this, I take the approach to make these changes out of habit.  I add `virtual` destructors to base classes and to use rule of 5 to ensure there will be move semantics on classes if I add a copy constructor/copy assignment/ or destructor.

So no, it does not _need_ to be specified, in particular because this is a rather simple use case.  I was just following my own personal practice..

Reply:
(and to your point, I missed following this for `FBTarget`, it originally wasn't a base class, which is why I missed it..)

Reply:
`final` is another similar situation.  I now add it out of habit to any child class that won't be derived from.  This is because there is a [use case where a child class can see a performance boost](https://devblogs.microsoft.com/cppblog/the-performance-benefits-of-final-classes/).

Reply:
Went ahead and removed it.